src/nanoemoji/write_font.py

```diff
@@ -576,23 +576,42 @@ def _colr_ufo(
     # We want to mutate our view of color glyphs
     color_glyphs = list(color_glyphs)
 
-    # Sort colors so the index into colors == index into CPAL palette.
     # We only store opaque colors in CPAL for COLRv1, as 'alpha' is
     # encoded separately.
-    colors = sorted(
-        set(
-            c if colr_version == 0 else c.opaque()
-            for c in chain.from_iterable(g.colors() for g in color_glyphs)
-            if not c.is_current_color()
-        )
+    colors = set(
+        c if colr_version == 0 else c.opaque()
+        for c in chain.from_iterable(g.colors() for g in color_glyphs)
+        if not c.is_current_color()
     )
+    # Sort colors so the index into colors == index into CPAL palette.
+    # Try to keep the original palette entry index if present.
+    iter_colors_without_index = iter(sorted(c for c in colors if c.index is None))
+    colors_by_index = {}
+    for c in colors:
+        if c.index is not None:
+            color = c.default()
```

Comment:
I don't find the behavior or name of default very intuitive. c.without_palette_index() or some such would at least tell me clearly what the intent here is.

Reply:
Yeah. I was thinking along the lines of "the default value if CPAL not supported".

Reply:
It's the default value part that throws me I think. In what sense does an rgba color have a default?

Reply:
It's a color var() that can have a default (which in turn is a color). Initially I went for a separate class for color variables, but then I would have to basically rewrite all the places where we accept Color to also accept a Color variables. It was just much easier to change Color to optionally carry a (palette entry) index.

Reply:
For me at least when reading the code a name that directly says what it does (c.rgba_only, c.without_palette_index, etc) would be clearer. That said, I'm fine if you want to leave the name default.

Reply:
changed to without_palette_index()